proxy-node-gateway/src/main/java/uk/gov/ida/notification/resources/EidasAuthnRequestResourceV2.java

```diff
@@ -0,0 +1,107 @@
+            VSPService vspService,
+            SamlFormViewBuilder samlFormViewBuilder) {
+        this.eidasSamlParserService = eidasSamlParserService;
+        this.vspService = vspService;
+        this.samlFormViewBuilder = samlFormViewBuilder;
+    }
+
+    @GET
+    @Path("/Redirect")
+    public View handleRedirectBinding(
+            @QueryParam(SamlFormMessageType.SAML_REQUEST) String encodedEidasAuthnRequest,
+            @QueryParam("RelayState") String relayState,
+            @Session HttpSession session) {
+        return handlePostBinding(encodedEidasAuthnRequest, relayState, session);
+    }
+
+    @POST
+    @Path("/POST")
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    public View handlePostBinding(
+            @FormParam(SamlFormMessageType.SAML_REQUEST) String encodedEidasAuthnRequest,
+            @FormParam(RelayState.DEFAULT_ELEMENT_LOCAL_NAME) String eidasRelayState,
+            @Session HttpSession session) {
+        return handleAuthnRequest(encodedEidasAuthnRequest, eidasRelayState, session);
+    }
+
+    private View handleAuthnRequest(String encodedEidasAuthnRequest, String eidasRelayState, HttpSession session) {
+        final EidasSamlParserResponse eidasSamlParserResponse = parseEidasRequest(encodedEidasAuthnRequest);
+        VSPAuthnRequestResponse vspResponse = generateHubRequestWithVSP();
+        logAuthnRequestInformation(session, eidasSamlParserResponse, vspResponse);
+        setResponseDataInSession(session, eidasSamlParserResponse, vspResponse);
+        return buildSamlFormView(vspResponse, eidasRelayState);
+    }
+
+    private void setResponseDataInSession(HttpSession session, EidasSamlParserResponse eidasSamlParserResponse, VSPAuthnRequestResponse vspResponse) {
+        session.setAttribute(SESSION_KEY_EIDAS_REQUEST_ID, eidasSamlParserResponse.getRequestId());
+        session.setAttribute(SESSION_KEY_EIDAS_CONNECTOR_PUBLIC_KEY, eidasSamlParserResponse.getConnectorPublicEncryptionKey());
+        session.setAttribute(SESSION_KEY_EIDAS_DESTINATION, eidasSamlParserResponse.getDestination());
+        session.setAttribute(SESSION_KEY_HUB_REQUEST_ID, vspResponse.getRequestId());
+    }
+
+    private EidasSamlParserResponse parseEidasRequest(String encodedEidasAuthnRequest) {
+        return eidasSamlParserService.parse(new EidasSamlParserRequest(encodedEidasAuthnRequest));
+    }
+
+    private VSPAuthnRequestResponse generateHubRequestWithVSP() {
+        return vspService.generateAuthnRequest();
+    }
+
+    private SamlFormView buildSamlFormView(VSPAuthnRequestResponse vspResponse, String relayState) {
+        URI hubUrl = vspResponse.getSsoLocation();
+        String samlRequest = vspResponse.getSamlRequest();
+        return samlFormViewBuilder.buildRequest(hubUrl.toString(), samlRequest, SUBMIT_BUTTON_TEXT, relayState);
+    }
+
+    private void logAuthnRequestInformation(HttpSession session, EidasSamlParserResponse eidasSamlResponse, VSPAuthnRequestResponse vspResponse) {
```

Comment:
Might be worth changing `eidasSamlResponse` to `eidasSamlParserResponse` just to disambiguate it from the eidas saml response we'll be sending back to the connector later in the journey.